metaflow/plugins/kfp/tests/test_kfp_s3_sensor.py

```diff
@@ -0,0 +1,42 @@
+from metaflow.plugins.kfp.kfp_s3_sensor import wait_for_s3_path
+
+from unittest.mock import call, Mock, patch, PropertyMock
+import pytest
+
+import boto3
+from botocore.exceptions import ClientError
+
+import base64
+import marshal
+
+"""
+To run these tests from your terminal, go to the root directory and run:
+
+`python -m pytest metaflow/plugins/kfp/tests/test_kfp_s3_sensor.py -c /dev/null`
+
+The `-c` flag above tells PyTest to ignore the setup.cfg config file which is used
+for the integration tests.
+
+"""
+
+
+def identity_formatter(path: str, flow_parameters: dict) -> str:
+    return path
+
+
+# This test ensures a timeout exception is raised when wait_for_s3_path
+# looks for a nonexistent S3 path. This ensures we don't have an idle
+# pod using up resources continuously.
+def test_wait_for_s3_path_timeout_exception():
```

Comment:
Can you also add unit test where the timeout error is not triggered? `moto` (https://github.com/spulec/moto) is a good package for mocking s3.